atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -164,3 +166,20 @@ def run(self):
         self.log.info("setting %s=%s", release_label, next_release)
         # Write the label back to the file (this is a property setter)
         dockerfile_labels[release_label] = next_release
+
+        if not self.predeclare_build:
+            return
+
+        nvr_data = {
+            'name': component,
+            'version': version,
+            'release': dockerfile_labels[release_label]
+        }
+
+        try:
+            pre_declared = self.xmlrpc.CGInitBuild(PROG, nvr_data)
+        except Exception as exc:
+            raise RuntimeError('unable to pre-declare build {}: {}'.format(nvr_data, exc))
```

Comment:
Another thing here: if CGInitBuild fails, don't we want to try the next release?

Can we distinguish between 'this n-v-r was already declared' and 'random failure'?

Reply:
#If build already exists, init_build will raise GenericError
so I can do that, even tho there is really just small window, within like 1 sec, that could happen

Reply:
I'm particularly thinking of the case where a build fails, after having pre-declared its n-v-r. The next time that build is submitted, will we pick the same release, or skip over the pre-declared one?

Reply:
skip predeclared, getNextRelease will give us simply just new number

Reply:
I'll retry for that 1sec window